### DISCUSSION TITLE
How to avoid the stucking upon lock.

### DISCUSSION THREAD

**Discussion by MarcoGonzalezM** [hsahovic/poke-env#546] [Q&A]
I've been trying to train an AI model for Pokémon using Pokémon Showdown. Pokémon Showdown lock from chat users when bot is detected. When Poké-env reads the warning from the lock it gets stuck and do not battle with the Player.battle_against() method, despite of the lock does not restrict making battles.

**Comment by hsahovic** [comment]
Hi @MarcoGonzalezM,
Are you running your script on a local instance of Pokémon Showdown, or the live website?

**Reply by MarcoGonzalezM** [reply]
It's on the live website. I know this would no happen on a local instance, but the issue is that the Script should be able to play, but it gets locked. As far as i have looked up it seems to be due to the Poké-env method trying to submit a message on the battle, but i am not sure about this.

**Reply by hsahovic** [reply]
Did you get an error message / exception logged?

**Reply by MarcoGonzalezM** [reply]
Yes, I receive a warning message:
WARNING - Trying to login as [Username], showdown returned ‽Guest [GuestNumber] - this might prevent future actions from this agent. Changing the agent's username might solve this problem.

[Username] and [GuestNumber] are my personal data of Pokémon Showdown. I prefer not to publish them. Also the ‽ symbol means that the account is locked from chat in case this would help.

**Reply by MarcoGonzalezM** [reply]
I dug into the problem a little bit. The issue occurs on the instruction await self.ps_client.logged_in.wait() in method _send_challenges() in class player.

I suppose that function is ensuring the profile is logged in before trying to start a battle.
